keps/sig-node/3953-node-resource-hot-plug/README.md

```diff
@@ -0,0 +1,659 @@
+
+				// Resync the resource managers
+				if err := kl.ResyncComponents(machineInfo); err != nil {
+					klog.ErrorS(err, "Error resyncing the kubelet components with machine info")
+				}
+			}
+		}
+	}
+```
+
+**Changes to resource managers to adapt to dynamic scale up of resources**
+
+1. Adding ResyncComponents() method to ContainerManager interface
+```go
+    // Manages the containers running on a machine.
+    type ContainerManager interface {
+        .
+        .
+        // ResyncComponents will resyc the resource managers like cpu, memory and topology managers
+	// with updated machineInfo
+	ResyncComponents(machineInfo *cadvisorapi.MachineInfo) error
+	.
+	.
+    )
+```
+
+2. Adding a method Sync to all the resource managers and will be invoked once there is dynamic resource change.
+
+```go
+        // Sync will sync the CPU Manager with the latest machine info
+	Sync(machineInfo *cadvisorapi.MachineInfo) error
+```
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Unit tests
+
+1. Add necessary tests in kubelet_node_status_test.go to check for the node status behaviour with dynamic node scale up.
+2. Add necessary tests in kubelet_pods_test.go to check for the pod cleanup and pod addition workflow.
+3. Add necessary tests in eventhandlers_test.go to check for scheduler behaviour with dynamic node capacity change.
+4. Add necessary tests in resource managers to check for managers behaviour to adopt dynamic node capacity change.
+
+
+##### e2e tests
+
+Following scenarios need to be covered:
+
+* Node resource information before and after resource hot plug.
+* State of Pending pods due to lack of resources after resource hot plug.
+* Resource manager states after the resynch of components.
+
+### Graduation Criteria
+
+
+#### Phase 1: Alpha (target 1.33)
+
+
+* Feature is disabled by default. It is an opt-in feature which can be enabled by enabling the `NodeResourceHotPlug`
+  feature gate.
+* Support the basic functionality for scheduler to consider pod-level resource requests to find a suitable node.
+* Support the basic functionality for kubelet to translate pod-level requests/limits to pod-level cgroup settings.
+* Unit test coverage.
+* E2E tests.
+* Documentation mentioning high level design.
+
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
```

Comment:
Please remove comments and populate these sections

Reply:
Updated it. Thank you for the review.